### DISCUSSION TITLE
soft delete와 연관관계 메서드의 필요성에 대해 헷갈리는 부분이 있습니다.

### DISCUSSION THREAD

**Discussion by Kwonminwoo** [so2zy/so2zy_BE#34] [Q&A]
연관관계 메서드의 필요성 및 soft delete 에 대해 궁금한 게 있습니다.
---
![image](https://github.com/so2zy/so2zy_BE/assets/34360434/4e4cc193-f48a-416c-89e7-63d16284a657)
예약 기능을 구현 하던 중 장바구니를 비워주기 위한 로직을 구현하게 되었습니다.
이 때 직접 repository의 장바구니에 담긴 room 정보를 지우게끔 구현했는데,
이 로직 이후 연관관계 편의 메서드를 이용해서 장바구니의 객실_장바구니 리스트에서도 데이터를 지워줘야 하는건가요? 

제가 생각하기에는 이후 장바구니를 사용하는 로직이 있다면 list에서도 지워야 하고 그렇지 않다면 굳이 지워야 할 필요가 없다고 생각하는데 제가 생각하는 것이 맞을까요

그리고 soft delete를 적용하기 위해서 auditing 엔티티의 deleted_at 필드들 활용하는 것인가요?
soft delete를 적용한다면 모든 삭제 로직은 soft로 처리되어야 하는 것인지에 대해서도 조금 헷갈립니다.
장바구니에 데이터를 담았다 뺐다 하는 로직도 soft로 처리되어야 할 필요가 있는건가요?

**Comment by Nine-JH** [comment]
1. `@OneToMany`의 `orphanRemoval` 옵션을 이용하면 굳이 roomcart 까지 가지 않고 cart에서 비워버릴 수 있습니다.
2. 장바구니는 softDelete 할 필요가 없을 것 같아요.

**Reply by Kwonminwoo** [reply]
아 맞네요!! 이해했습니다 감사합니다
